node/services/userService.js

```diff
@@ -167,11 +167,18 @@ module.exports = {
         }
     },
 
-    deleteUser: (req, res, next) => {
-        UserModel.remove({ _id: ObjectId(req.params.user_id) })
-            .then(res.status(204).send({ msg: 'deleted' }))
-            .catch(error => {
-                return res.status(error.status || 500).send(error);
-            });
+    deleteUser: async (req, res, next) => {
+        const { user_id: userID } = req.params;
+
+        if (_.isNil(userID)) {
+            return res.status(400).send(`Invalid user_id submitted: ${userID}`);
+        }
+
+        try {
+            await UserModel.remove({ _id: ObjectId(req.params.user_id) });
+            res.status(204).send({ msg: 'deleted' });
+        } catch (err) {
+            res.status(err.status || 500).send(err);
```

Comment:
Will this log an error? I think you have to `throw res.status(...).send(...)` to get it to log the error. Not a show stopper though.

Reply:
No, throwing will actually cause a problem. We don't want the error to throw - we want to return a response with an error code. I think throwing will potentially crash the app - right now the app logs it as an uncaught exception, but some versions of node (?) may not do the same.

Reply:
Interesting. Just not the behavior I've seen but that's fine. We just have no logging at all so if we were curious as to errors we'd have no visibility

Reply:
We can add console logs, if that's a concern - or implement something more advanced, that's just the easiest way. Where have you seen errors been thrown as responses before? I'd be really curious to see how that works. I'm not super familiar with the ins-and-outs of returning responses from node servers, I just saw the error being logged as an uncaught exception on the service, and the warning about how this isn't always handled (uncaught exceptions crash some of our CBT apps), so I thought we should get rid of it.